chain-impl-mockchain/src/ledger/ledger.rs

```diff
@@ -1610,6 +1622,39 @@ impl Ledger {
     }
 }
 
+impl ApplyBlockLedger {
+    pub fn apply_fragment(&self, fragment: &Fragment) -> Result<Self, Error> {
+        let ledger = self
+            .ledger
+            .apply_fragment(&self.ledger_params, fragment, self.block_date)?;
+        Ok(ApplyBlockLedger {
+            ledger,
+            ..self.clone()
+        })
+    }
+
+    pub fn finish(self, maybe_gp_content: Option<&HeaderGPContentEvalContext>) -> Ledger {
```

Comment:
The maybe_gp_content parameter type could may be an enum like

```
enum ConsensusEvalContext {
    Bft,  // nothing to see here, move along
    Praos {
        // ...
    }
}
```

To make it less ad-hoc and provide a pattern for extension in the future.

Reply:
done